app/src/main/java/org/kontalk/util/MessageUtils.java

```diff
@@ -708,17 +735,29 @@ public static String toString(byte[] text) {
         return text;
     }
 
-    public static void convertSmileys(Editable input) {
+    public static boolean convertSmileys(Editable input) {
```

Comment:
We need a boolean here to decide if we should restart the IME or not. Restarting the IME too often leads to unpredictable behaviour. But if we don't restart the IME there are problems with already replaced emojis (e.g. https://github.com/kontalk/androidclient/pull/842#issuecomment-255823501). So I only restart the IME if I know that some ASCII-characters were replaced, and therefore I need the boolean.